spark/gcc-output-parser
# Add support for node@16, npm@8 and circleci

## Description
Makes the required changes so the library works with `node@16x` and `npm@8.x`. This PR also migrates from Github Actions to CircleCI and introduces the same automatic publishing flow as [`mongoose-models`](https://github.com/particle-iot/mongoose-models#releasing-changes).

## How to Test

1. Pull the branch: `chore/sc-104116/update-gcc-output-parser-to-node-js-v16-and`
2. Update your local node version to `v16.x` and npm to version to `v8.x`. Use `nvm use 16` or equivalent.
3. Verify your updates: `node -v && npm -v`
4. Install dependencies: `npm i`
5. Run the tests `npm run test`

**Outcome**
All tests should pass. npm i should run without issues.

## Related / Discussions


- https://app.shortcut.com/particle/story/104116/update-gcc-output-parser-to-node-js-v16-and-npm-v8

## Completeness
